tests/streamlit_ui_tests/streamlit.spec.ts

```diff
@@ -1,9 +1,302 @@
+  return mito.locator('.endo-column-header-container').locator('div').filter({ hasText: columnName }).first();
+}
+
+
+
+test('Can render Mito spreadsheet', async ({ page }) => {
+  const mito = await getMitoFrame(page);
+  await mito.getByRole('button', { name: 'Import Files' }).click();
+  await expect(mito.getByText('test.csv')).toBeVisible();
+});
+
+
+test.describe('Home Tab Buttons', () => {
+  test('Import CSV File (Double Click)', async ({ page }) => {
+    const mito = await getMitoFrame(page);
+    await mito.getByRole('button', { name: 'Import Files' }).click();
+    await mito.getByText('test.csv').dblclick();
+    await expect(mito.getByTitle('Column1')).toBeVisible();
+  });
+
+  test('Copy Button', async ({ page }) => {
+    const mito = await getMitoFrameWithTestCSV(page);
+
+    await mito.getByTitle('Column1').click();
+    await clickButtonAndAwaitResponse(page, mito, 'Copy')
+
+    // TODO: There are some bugs with Playwrite 
+    // https://github.com/microsoft/playwright/issues/18901
+    // That make it hard to check the clipboard contents
+    // But we can check that this has a black border
+    // Check that the endo-column-header-final-container has, among it's style, 
+    // the text 'border-top: 1px dashed black;'
+    await expect(mito.locator('.endo-column-header-final-container').first()).toHaveAttribute('style', /border-top: 1px dashed black;/);
+  });
+
+  test('Formatting Butons', async ({ page }) => {
+    const mito = await getMitoFrameWithTestCSV(page);
+
+    await clickButtonAndAwaitResponse(page, mito, 'Format all of the selected columns as currency. This only changes the display of the data, and does not effect the underlying dataframe.')
+    await expect(mito.getByText('$1')).toBeVisible();
+    await clickButtonAndAwaitResponse(page, mito, 'Format all of the selected columns as percentage. This only changes the display of the data, and does not effect the underlying dataframe.')
+    await expect(mito.getByText('100%')).toBeVisible();
+    await clickButtonAndAwaitResponse(page, mito, 'Increase the number of decimal places that are displayed in the selected number columns.')
+    await expect(mito.getByText('100.0%')).toBeVisible();
+    await clickButtonAndAwaitResponse(page, mito, 'Decrease the number of decimal places that are displayed in the selected number columns.')
+    await expect(mito.getByText('100%')).toBeVisible();
+    await mito.getByText('100%').click();
+  });
+
+  test.skip('Conditional Formatting', async ({ page }) => {
+    const mito = await getMitoFrameWithTestCSV(page);
+    
+    await clickButtonAndAwaitResponse(page, mito, 'Conditional Formatting')
+    
+    await checkOpenTaskpane(mito, 'Conditional Formatting');
+    
+    // This test is currently broken, for some reason this button won't click
+    await mito.getByRole('button', { name: 'Add Conditional Formatting Rule' }).click();
```

Comment:
Given the tests are passing, I suppose this comment is out of date?

Reply:
Tbh, skipping this test currently. Struggled to get it working, so will circle back to it when we've got a bit more experience :)